test/conftest.py

```diff
@@ -12,29 +21,97 @@
+    server, server_thread = utilities.start_server()
+    yield
+    print("\nClosing server thread...")
+    server.shutdown()
+    server_thread.join()
 
-    if "pyonly" in request.config.invocation_params.args:
-        return
 
-    create_xpi()
+@pytest.fixture()
+def default_params(
+    tmp_path: Path, num_browsers: int = NUM_BROWSERS
+) -> Tuple[ManagerParams, List[BrowserParams]]:
+    """Just a simple wrapper around task_manager.load_default_params"""
 
-    print("Starting local_http_server")
-    server, server_thread = utilities.start_server()
+    manager_params = ManagerParams(
+        num_browsers=NUM_BROWSERS
+    )  # num_browsers is necessary to let TaskManager know how many browsers to spawn
+
+    browser_params = [
+        BrowserParams(display_mode="headless") for _ in range(NUM_BROWSERS)
+    ]
+    manager_params.data_directory = tmp_path
+    manager_params.log_directory = tmp_path
+    for i in range(num_browsers):
+        browser_params[i].display_mode = "headless"
+    return manager_params, browser_params
+
+
+@pytest.fixture()
+def task_manager_creator(
+    server,
+    xpi,
+) -> Callable[[Tuple[ManagerParams, List[BrowserParams]]], Tuple[TaskManager, Path]]:
+    """We create a callable that returns a TaskManager that has
+    been configured with the Manager and BrowserParams"""
+
+    def _create_task_manager(
+        params: Tuple[ManagerParams, List[BrowserParams]]
+    ) -> Tuple[TaskManager, Path]:
+        manager_params, browser_params = params
+        db_path = manager_params.data_directory / "crawl-data.sqlite"
+        structured_provider = SQLiteStorageProvider(db_path)
+        manager = TaskManager(
+            manager_params,
+            browser_params,
+            structured_provider,
+            None,
+        )
+        return manager, db_path
+
+    return _create_task_manager
+
+
+@pytest.fixture()
+def http_params(
+    default_params: Tuple[ManagerParams, List[BrowserParams]],
+) -> Callable[[str], Tuple[ManagerParams, List[BrowserParams]]]:
+    manager_params, browser_params = default_params
+    for browser_param in browser_params:
+        browser_param.http_instrument = True
+
+    def parameterize(
+        display_mode: str = "headless",
+    ) -> Tuple[ManagerParams, List[BrowserParams]]:
+        for browser_param in browser_params:
+            browser_param.display_mode = display_mode
+        return manager_params, browser_params
+
+    return parameterize
 
-    def local_http_server_stop():
-        print("\nClosing server thread...")
-        server.shutdown()
-        server_thread.join()
 
-    request.addfinalizer(local_http_server_stop)
+@pytest.fixture()
+def mp_logger(tmp_path: Path) -> Generator[MPLogger, Any, None]:
+    log_path = tmp_path / "openwpm.log"
+    logger = MPLogger(log_path, log_level_console=logging.DEBUG)
+    yield logger
+    logger.close()
+    # The performance hit for this might be unacceptable but it might help us discover bugs
```

Comment:
This costs us 15s per test. Is that too much?

Reply:
https://github.com/mozilla/OpenWPM/pull/753/checks?check_run_id=1838971116#step:12:2705

Reply:
What's the mean in terms of total test suite runtime? In general I think it's okay to trade off some test time if it means that we'll catch errors that we'd otherwise miss.

Reply:
Hmm, running a super unprofessional benchmark, the test suite running without this check was 20s slower :sweat_smile: 
29:03 (with this check) vs 29:30 (with this check commented out)